streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1452,4 +1459,24 @@ private RuntimeException performOnStreamTasks(final StreamTaskAction action) {
         return firstException;
     }
 
+    /**
+     * Return information about the current {@link StreamThread}.
+     *
+     * @return {@link ThreadMetadata}.
+     */
+    public synchronized ThreadMetadata threadMetadata() {
+        return threadMetadata;
+    }
+
+    private synchronized void updateThreadMetadata() {
+        Set<TaskMetadata> activeTasksMetadata = new HashSet<>();
```

Comment:
nit: add `final`

We try to use `final` where ever possible. Please also add below (including the for-loop).